doc/LibcurlTransportAdapter.md

```diff
@@ -0,0 +1,162 @@
+ 
+1.	Resolve DNS from request URL.
+2.	Get a network socket from the OS.
+3.	Stablish TCP connection to the server.
+4.	Secure connection with TLS when using HTTPS.
+ 
+So, only the first four steps are managed by libcurl. A connection channel is established to the server, but no message is sent. Libcurl returns from curl_easy_perform leaving the connection ready for users to start calling [curl_easy_send](https://curl.se/libcurl/c/curl_easy_send.html) and [curl_easy_recv](https://curl.se/libcurl/c/curl_easy_recv.html). 
+ 
+Most of the libcurl handle configuration becomes useless when the handle is changed to custom protocol. For example, it becomes irrelevant creating a headers list, or setting an HTTP method (GET, POST, etc.). Libcurl will ignore it, and it will be now the customer who takes connection ownership.  What this means for the libcurlTransportAdapter is that it has to learn how to speak raw HTTP and translate an Azure:: Core:: Http:: Request into a pure HTTP call.
+
+#### Receiving raw HTTP
+
+The most tedious task, while speaking raw HTTP, is reading and parsing responses. As mentioned before, the main purpose of using the poor man's substitute (custom protocol) to re-implement HTTP (even though libcurl provides HTTP support natively) is to enable an Azure SDK client to start a download operation and let customers to decide when to pull chunks of data from the response (directly from the network socket, without any intermediate buffers or using callback functions).
+ 
+An HTTP request is nothing but a string with a specific format, it has special text delimiters to distinguish between the request line, headers and body. As soon as the server receives and parses the request, it would trigger an operation and eventually produce a response. The response needs to be HTTP formatted using text delimiters to indicate where the headers section starts and where it ends. Then the server will start sending small parts of it. On the client side, calling _curl_easy_recv()_ will get any bytes that have arrived. As soon as the function pulls the bytes from the socket, the socket will receive more data that will be written on top of the previous data. The function _curl_easy_recv()_ is a way to tell the OS, _“Hey, I got this data, you can write on top of it now"_.
```

Comment:
One could argue that the same applies to ANY text based protocol (POP3, IMAP4, NNTP, FTP, SMTP, MIME, BASE64, PEM, etc) - they are all strings with a specific format.